windows/src/desktop/kmshell/main/initprog.pas

```diff
@@ -231,6 +231,7 @@ function Init(var FMode: TKMShellMode; KeyboardFileNames: TWideStrings; var FSil
       else if s = '-nowelcome'   then FNoWelcome := True
       else if s = '-kw' then FMode := fmKeyboardWelcome  // I2569
       else if s = '-kp' then FMode := fmKeyboardPrint  // I2329
+      else if s = '-log' then begin Inc(i); FLogFile := ParamStr(i); end
```

Comment:
Is it possible for `ParamStr()` to end in `-log` without a log file, that `ParamStr(i)` throws?

Reply:
`ParamStr(ParamCount+n)` always returns empty string

Reply:
So if there is no logfile passed, then it will effectively be a no-op. :)